ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -346,6 +346,10 @@ func (r *RayDashboardClient) SubmitJobReq(ctx context.Context, request *RayJobRe
 
 	body, _ := io.ReadAll(resp.Body)
 
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
```

Comment:
Is there any reason for adding this update? Did you find any edge cases?

Reply:
yes, I found that a 403 or any other response code is not logged propoerly because returned error is nil

Reply:
Thanks!